src/napari_plugin_manager/qt_package_installer.py

```diff
@@ -63,6 +64,28 @@ def _constraints_file(cls) -> str:
         return f.name
 
 
+class NapariUvInstallerTool(UvInstallerTool):
+    @staticmethod
+    def constraints() -> Sequence[str]:
+        """
+        Version constraints to limit unwanted changes in installation.
+        """
+        return [f'napari=={_napari_version}']
```

Comment:
The `uv pip install --upgrade` upgrade not only itself but also all its dependencies. But do not think about other packages installed in the environment. 

So if a package depends on NumPy and Numba is installed, having only pinned napari means that the upgrade may end with a non-working environment, as import of Numba crashes.

Reply:
Or is the suggestion to extend the `constraints()` list with all the dependencies of the currently installed `napari` version?

Reply:
@Czaki any word here?